bee-storage/bee-storage-rocksdb/src/access/insert.rs

```diff
@@ -9,14 +9,19 @@ use bee_message::{Message, MessageId};
 use bee_packable::Packable;
 use bee_storage::{access::Insert, system::System, StorageBackend};
 
-impl Insert<u8, System> for Storage {
-    fn insert(&self, key: &u8, value: &System) -> Result<(), <Self as StorageBackend>::Error> {
-        self.inner
-            // Packing to bytes can't fail.
-            .put_cf(self.cf_handle(CF_SYSTEM)?, [*key], value.pack_to_vec().unwrap())?;
+/// This would normally be `impl Insert<u8, System> for Storage` but there is no way to have a private trait impl and
+/// only the [`Storage`] itself should be able to insert system values.
```

Comment:
would it be possible to not directly implement it for `Storage` type, however for another type? 
This other type then is owned by the storage struct e.g.:
```
struct Storage {
  insert_u8: InsertU8,
}
```
and `Storage` just offers getter and setter?

Reply:
Not really because this method needs what's inside `Storage` so it needs to have access to it.

Reply:
and passing a reference of the stoarge?

Reply:
I do pass a reference to the storage

Reply:
okey unterstand, so maybe private traits will come at a later point in time